src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -2059,9 +2071,13 @@ module Make_str (A : Wire_types.Concrete) = struct
             let%bind state_body_hash =
               Mina_state.Protocol_state.Body.hash_checked state_body
             in
+            let global_slot =
+              Mina_state.Protocol_state.Body.consensus_state state_body
+              |> Consensus.Data.Consensus_state.global_slot_since_genesis_var
```

Comment:
This is the `global_slot` from the *old* state, not the current block's one.

Reply:
Thanks for catching the bug!